src/clr-angular/forms/datepicker/day.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+
+import {Component, Input} from "@angular/core";
+
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+
+import {DayViewModel} from "./model/day-view.model";
+import {DayModel} from "./model/day.model";
+import {DateIOService} from "./providers/date-io.service";
+import {DateNavigationService} from "./providers/date-navigation.service";
+
+@Component({
+    selector: "clr-day",
+    template: `
+        <button
+            class="day-btn"
+            type="button"
+            [class.is-today]="dayView.isTodaysDate"
+            [class.is-disabled]="dayView.isDisabled"
+            [class.is-selected]="dayView.isSelected"
+            [attr.tabindex]="dayView.tabIndex"
+            (click)="setDay(dayView)"
+            (focus)="onDayViewFocus(dayView)">
+            {{dayView.dayModel.date}}
+        </button>
+    `,
+    host: {"[class.day]": "true"}
+})
+export class ClrDay {
+    constructor(private _dateNavigationService: DateNavigationService, private _dateIOService: DateIOService,
+                private _ifOpenService: IfOpenService) {}
+
+    /**
+     * DayViewModel input which is used to build the Day View.
+     */
+    @Input("clrDayView") dayView: DayViewModel;
+
+    /**
+     * Updates the focusedDay in the DateNavigationService when the ClrDay is focused.
+     */
+    onDayViewFocus(dayView: DayViewModel) {
+        this._dateNavigationService.focusedDay = dayView.dayModel;
+    }
+
+    /**
+     * Updates the selectedDay when the ClrDay is selected and closes the datepicker popover.
+     */
+    setDay(dayView: DayViewModel): void {
```

Comment:
`setDay()` is too confusing a name, my first reflex was "I can change the day model of a day component?"

A better name would be `selectDay()`, or something similar.

Reply:
fixed